app/build.gradle

```diff
@@ -753,3 +753,10 @@ if (gradle.hasProperty('localProperties.autoPublish.application-services.dir'))
     ext.appServicesSrcDir = gradle."localProperties.autoPublish.application-services.dir"
     apply from: "../${appServicesSrcDir}/build-scripts/substitute-local-appservices.gradle"
 }
+
+tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
+    kotlinOptions.freeCompilerArgs += [
+            "-Xuse-experimental=kotlinx.coroutines.ExperimentalCoroutinesApi",
```

Comment:
@NotWoods I went through the app:dependencies while investigating this PR and saw that we end up bumping the kotlin version to 1.3.72 implicitly. Curious if bumping the coroutines version could help resolve some of these annotation problems?

Reply:
We do this in AC as well (albeit on a per-component basis), as many Android Components use `ExperimentalCoroutinesApi`. However, the newer Kotlin version does make this much easier to avoid. 

The current annotation forces any caller to use it, and their callers, and their callers, winding all the way up the call stack.

Newer Kotlin versions (1.7 IIRC) let you just write `OptIn(ExperimentalCoroutinesApi)` and will just annotate that call, without putting requirements on its callers.

Since we've cut a release branch we can update the Kotlin version then pull those changes in here.

Reply:
@liuche Sure!

> Is it because you'd need to annotate the detekt config? Could you help me understand why this is necessary, or what other options there are? (Is the alternative to annotate each usage?)

I guess applying `-Xuse` was agreed in [this thread](https://github.com/mozilla-mobile/fenix/pull/9705#issuecomment-637889887).

Alternatively we can annotate each usage with `@UseExperimental` to avoid compiler warnings that breaks our build (`allWarningsAsErrors=true`).

Reply:
Ah, you're totally right. I thought at the time that it would be simpler to not bump the Kotlin version, but when looking at other dependencies, it seems like we already use 1.3.72, so now that I've dug a little more into it, option a) seems to be better, to switch to @OptIn or @UseExperimental.

Sorry for all the back and forth.

Reply:
Not a problem! Will rebase this PR with @NotWoods changes.